code/modules/power/lighting.dm

```diff
@@ -340,6 +340,27 @@
 	var/area/A = get_area(src)
 	return A.lightswitch && A.power_light
 
+/obj/machinery/light/proc/set_flicker(amount, flicker_time_lower, flicker_time_upper)
+	var/theflicker
+	if(flickering)
+		deltimer(theflicker)
+		resetflickers()
+	if(!flickering)
+		random_flicker = TRUE
+		flicker(TRUE)
+		if(flicker_time_lower)
+			flicker_time_lower_min = flicker_time_lower
+		if(flicker_time_upper)
+			flicker_time_upper_max = flicker_time_upper
+		theflicker = addtimer(CALLBACK(src, PROC_REF(resetflickers), amount))
+
+/obj/machinery/light/proc/resetflickers()
```

Comment:
Doc missing, also is this a port of teh tg proc? I think they had one

Reply:
No i made my own.